api/components.go

```diff
@@ -0,0 +1,27 @@
+package api
+
+import (
+	"github.com/cloudtrust/keycloak-client/v2"
+	"gopkg.in/h2non/gentleman.v2/plugins/body"
+	"gopkg.in/h2non/gentleman.v2/plugins/url"
+)
+
+const (
+	// API Keycloak out-of-the-box
+	kcComponentsPath  = "/auth/admin/realms/:realm/components"
+	kcComponentIDPath = kcComponentsPath + "/:id"
+)
+
+// GetComponent get the representation of the component.
+func (c *Client) GetComponent(accessToken string, realmName, componentID string) (keycloak.ComponentRepresentation, error) {
```

Comment:
I guess you would prefer to have GetComponents and filter on provider_type/provider_id. You initially won't know the component ID.
I imagine a ComponentTool that you instanciate with provider_type/provider_id/configName.
You would have these functions:
- GetComponentConfigKeyValue(realm) (ComponentConfigKeyValue, error)
- NewComponentConfigKeyValue(realm)
- UpdateComponentConfigKeyValue(ComponentConfigKeyValue) error
Functions for ComponentConfigKeyValue:
- UnmarshalJSONEntry(key string, reference any) error
- MarshalJSONEntry(key string, value any) error
- DeleteEntry(key string) bool